backend/corpora/common/entities/dataset.py

```diff
@@ -1,9 +1,106 @@
 from .entity import Entity
-from ..corpora_orm import DbDataset
+from ..corpora_orm import DbDataset, DbDatasetArtifact, DbDeploymentDirectory, DbContributor, DbDatasetContributor
 
 
 class Dataset(Entity):
     table = DbDataset
 
     def __init__(self, db_object: DbDataset):
         super().__init__(db_object)
+
+    @classmethod
+    def create(
+        cls,
+        revision: int = 0,
+        name: str = "",
+        organism: str = "",
+        organism_ontology: str = "",
+        tissue: str = "",
+        tissue_ontology: str = "",
+        assay: str = "",
+        assay_ontology: str = "",
+        disease: str = "",
+        disease_ontology: str = "",
+        sex: str = "",
+        ethnicity: str = "",
+        ethnicity_ontology: str = "",
+        source_data_location: str = "",
+        preprint_doi: str = "",
+        publication_doi: str = "",
+        artifacts: list = None,
+        contributors: list = None,
+        deployment_directories: list = None,
+    ) -> "Dataset":
+        """
+        Creates a new dataset and related objects
+        :param revision:
+        :param name:
+        :param organism:
+        :param organism_ontology:
+        :param tissue:
+        :param tissue_ontology:
+        :param assay:
+        :param assay_ontology:
+        :param disease:
+        :param disease_ontology:
+        :param sex:
+        :param ethnicity:
+        :param ethnicity_ontology:
+        :param source_data_location:
+        :param preprint_doi:
+        :param publication_doi:
+        :param artifacts:
+        :param contributors:
+        :param deployment_directories:
+        :return:
+        """
+        uuid = cls.db.generate_id(DbDataset)
+
+        """
+        Prevent accidentally linking an existing row to a different Dataset. This maintains the relationship of one
+        to many for artifacts and deployment_directories
+        """
+        artifacts = artifacts if artifacts else []
+        deployment_directories = deployment_directories if deployment_directories else []
+        contributors = contributors if contributors else []
+
+        [artifact.pop("id", None) for artifact in artifacts]  # sanitize of ids
+        [deployment_directory.pop("id", None) for deployment_directory in deployment_directories]  # sanitize of ids
+
+        new_db_object = DbDataset(
+            id=uuid,
+            revision=revision,
+            name=name,
+            organism=organism,
+            organism_ontology=organism_ontology,
+            tissue=tissue,
+            tissue_ontology=tissue_ontology,
+            assay=assay,
+            assay_ontology=assay_ontology,
+            disease=disease,
+            disease_ontology=disease_ontology,
+            sex=sex,
+            ethnicity=ethnicity,
+            ethnicity_ontology=ethnicity_ontology,
+            source_data_location=source_data_location,
```

Comment:
This isn't really part of this PR, but what does "source_data_location" represent for a dataset? It's currently only in the schema at the project level.

Reply:
I don't have an answer. @maniarathi any clues?

Reply:
@mckinsel @Bento007 This is to keep track of the original data link. We'll likely supporting transferring files from a drive link or dropbox link that the users give us, so this would keep track of that.